File: CS/WebApplicationCS/Controllers/HomeController.cs
using DevExpress.Web.Mvc;
using DevExpress.XtraRichEdit;
using System;
using System.Text;
using System.Web.Mvc;
using WebApplicationCS.Models;

namespace WebApplicationCS.Controllers {
    public class HomeController : Controller {
		
        public ActionResult Index() {
            return View();
        }

        [HttpPost, ValidateInput(false)]
        public ActionResult RichEditPartial(string documentID) {
            RichEditModel model = new RichEditModel();
            model.documentID = documentID;
            model.rtfContent = Encoding.UTF8.GetString(RichEditExtension.SaveCopy("richEdit", DocumentFormat.Rtf));
            return PartialView("_RichEditPartial", model);
        }

        [ValidateInput(false)]
        public ActionResult GridViewPartial() {
            var model = OleDbDataProvider.GetCars();
            return PartialView("_GridViewPartial", model);
        }

        [HttpPost, ValidateInput(false)]
        public ActionResult GridViewPartialAddNew(Car item) {
            if (ModelState.IsValid) {
                try {
                    item.RtfContent = Encoding.UTF8.GetString(RichEditExtension.SaveCopy("richEdit", DocumentFormat.Rtf));
                    //Note that data modifications are not allowed in online demos. 
                    //To allow editing in local/offline mode, download the example 
                    //and comment out the line below and in GridViewPartialUpdate action method respectively.

                    //OleDbDataProvider.AddNewItem(item);
                    ViewData["EditError"] = "Data modifications are not allowed in online demos";
                }
                catch (Exception e) {
                    ViewData["EditError"] = e.Message;
                }
            }
            else
                ViewData["EditError"] = "Please, correct all errors.";           
            var model = OleDbDataProvider.GetCars();
            return PartialView("_GridViewPartial", model);
        }

        [HttpPost, ValidateInput(false)]
        public ActionResult GridViewPartialUpdate(Car item) {
            if (ModelState.IsValid) {
                try {
                    item.RtfContent = Encoding.UTF8.GetString(RichEditExtension.SaveCopy("richEdit", DocumentFormat.Rtf));
                    //OleDbDataProvider.UpdateItem(item);
                    ViewData["EditError"] = "Data modifications are not allowed in online demos";
                }
                catch (Exception e) {
                    ViewData["EditError"] = e.Message;
                }
            }
            else
                ViewData["EditError"] = "Please, correct all errors.";
            var model = OleDbDataProvider.GetCars();
            return PartialView("_GridViewPartial", model);
        }

        [HttpPost, ValidateInput(false)]
        public ActionResult GridViewPartialDelete(Int32 ID) {
            if (ID >= 0) {
                try {
                    OleDbDataProvider.DeleteItem(ID);
                }
                catch (Exception e) {
                    ViewData["EditError"] = e.Message;
                }
            }
            var model = OleDbDataProvider.GetCars();
            return PartialView("_GridViewPartial", model);
        }
    }
}

File: VB/WebApplicationVB/Controllers/HomeController.vb
Imports DevExpress.Web.Mvc
Imports DevExpress.XtraRichEdit

Public Class HomeController
	Inherits System.Web.Mvc.Controller

	Public Function Index() As ActionResult
		Return View()
	End Function

	<HttpPost, ValidateInput(False)>
	Public Function RichEditPartial(ByVal documentID As String) As ActionResult
		Dim model As New RichEditModel()
		model.documentID = documentID
		model.rtfContent = Encoding.UTF8.GetString(RichEditExtension.SaveCopy("richEdit", DocumentFormat.Rtf))
		Return PartialView("_RichEditPartial", model)
	End Function

	<ValidateInput(False)>
	Public Function GridViewPartial() As ActionResult
		Dim model = OleDbDataProvider.GetCars()
		Return PartialView("_GridViewPartial", model)
	End Function

	<HttpPost, ValidateInput(False)>
	Public Function GridViewPartialAddNew(ByVal item As Car) As ActionResult
		If ModelState.IsValid Then
			Try
				item.RtfContent = Encoding.UTF8.GetString(RichEditExtension.SaveCopy("richEdit", DocumentFormat.Rtf))
				'Note that data modifications are not allowed in online demos. 
				'To allow editing in local/offline mode, download the example 
				'and comment out the line below and in GridViewPartialUpdate action method respectively.

				'OleDbDataProvider.AddNewItem(item);
				ViewData("EditError") = "Data modifications are not allowed in online demos"
			Catch e As Exception
				ViewData("EditError") = e.Message
			End Try
		Else
			ViewData("EditError") = "Please, correct all errors."
		End If
		Dim model = OleDbDataProvider.GetCars()
		Return PartialView("_GridViewPartial", model)
	End Function

	<HttpPost, ValidateInput(False)>
	Public Function GridViewPartialUpdate(ByVal item As Car) As ActionResult
		If ModelState.IsValid Then
			Try
				item.RtfContent = Encoding.UTF8.GetString(RichEditExtension.SaveCopy("richEdit", DocumentFormat.Rtf))
				'OleDbDataProvider.UpdateItem(item);
				ViewData("EditError") = "Data modifications are not allowed in online demos"
			Catch e As Exception
				ViewData("EditError") = e.Message
			End Try
		Else
			ViewData("EditError") = "Please, correct all errors."
		End If
		Dim model = OleDbDataProvider.GetCars()
		Return PartialView("_GridViewPartial", model)
	End Function

	<HttpPost, ValidateInput(False)>
	Public Function GridViewPartialDelete(ByVal ID As Int32) As ActionResult
		If ID >= 0 Then
			Try
				OleDbDataProvider.DeleteItem(ID)
			Catch e As Exception
				ViewData("EditError") = e.Message
			End Try
		End If
		Dim model = OleDbDataProvider.GetCars()
		Return PartialView("_GridViewPartial", model)
	End Function

End Class